File: python/2492.Minimum-Score-of-a-Path-Between-Two-Cities.py
class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        adj_list = defaultdict(list)
        visited = set()
        result = float("inf")

        for src, dst, dist in roads:
            adj_list[src].append((dst, dist))
            adj_list[dst].append((src, dist))

        def dfs(node):
            nonlocal result
            if node in visited:
                return
            visited.add(node)
            for dst, dist in adj_list[node]:
                result = min(result, dist)
                dfs(dst)

        dfs(1)

        return result

File: javascript/2492.Minimum-Score-of-a-Path-Between-Two-Cities.js
/**
 * @param {number} n
 * @param {number[][]} roads
 * @return {number}
 */
var minScore = function (n, roads) {
  let result = Infinity;
  const adj_list = new Map();
  const visited = new Set();

  for (const [src, dst, dist] of roads) {
    if (!adj_list.has(src)) adj_list.set(src, []);
    if (!adj_list.has(dst)) adj_list.set(dst, []);

    adj_list.get(src).push([dst, dist]);
    adj_list.get(dst).push([src, dist]);
  }

  const dfs = (node) => {
    if (visited.has(node)) {
      return;
    }

    visited.add(node);

    for (const [dst, dist] of adj_list.get(node)) {
      result = Math.min(result, dist);
      dfs(dst);
    }
  };

  dfs(1);

  return result;
};
